chapters/technology-research/architecture.tex

```diff
@@ -0,0 +1,197 @@
+The ultimate goal is to minimize the lifetime cost of the system
+and to maximize programmer productivity.}}}
+\cite{martin_clean_architecture}
+
+Dle zdroje \cite{architecture} lze rozdělit architektury do dvou skupin:
+architektury zaměřené na databázi a architektury zaměřené na doménu.
+V následujících sekcích budou popsány rozdíli mezi oběmi skupinami
+a vybrané konkrétní případy architektur.
+
+\subsection{Architektury zaměřené na databázi}
+
+Architektury zaměřené na databázi,
+také nazývané \emph{Database-centric architectures},
+byly dle \cite{architecture} prvním typem softwarových architektur.
+Tyto architektury staví databáze do středu dění,
+jak jde vidět z obrázku \ref{fig:architecture_database}.
+
+\begin{figure}
+    \centering
+    \includegraphics[width=0.5\linewidth]{assets/technology-research/architecture/database-centric.jpg}
+    \caption{Schéma architektury zaměřené na databázi \todo{do vektoru} \cite{architecture}}
+    \label{fig:architecture_database}
+\end{figure}
+
+Příkladem tohoto typu architektury je tradiční 3-vrstvá architektura.
+Ta je robustní a škálovatelná.
+Skládá se ze tří vrstev.
+První vrstva,
+prezentační vrstva,
+obsahuje tvorbu UI,
+bez samotného kódu s logikou.
+Druhá vrstva,
+aplikační vrstva,
+obsahuje samotné kód s logikou.
+Tato vrstva by měla být nezávislá na prezentační vrstvě,
+avšak má explicitní závislost na datové vrstvě.
+Poslední vrstva,
+datová vrstva,
+je nejspodnější vrstvou,
+která se stará o manipulaci a tok dat z databáze do nižších vrstev.
+\cite{architecture}
+
+\begin{listing}
+    \caption{Ukázka přístupu zaměřeného na databázi v jazyce Java \cite{architecture}}
+    \label{code:architecture-database}
+    \begin{minted}{java}
+package ua.com.crosp.testapp.domain;
+import ua.com.crosp.testapp.datalayer.PostsRepository;
+
+public class GetRecentPostsUseCase
+implements GetRecentPostsUseCaseContract {
+    private PostsRepository mPostsRepository;
+
+    public Single<Post.List> execute(Params params) {
+        // Execute
+    }
+}
+    \end{minted}
+\end{listing}
+
+V této architektuře je největší role přikládána databázi
+a často je tak datová a aplikační vrstva úzce propojena.
+V nejhorších případech na datové vrstvě závisí i vrstva prezentační.
+\cite{architecture}
+Ukázka přístupu pomocí této architektury je znázorněna ve výpisu kódu
+\ref{code:architecture-database}.
+
+\subsection{Architektury zaměřené na doménu}
+
+\begin{figure}
+    \centering
+    \includegraphics[width=0.5\linewidth]{assets/technology-research/architecture/domain-centric.jpeg}
+    \caption{Schéma architektury zaměřené na doménu \todo{do vektoru; oříznout?} \cite{architecture}}
+    \label{fig:architecture_domain}
+\end{figure}
+
+Architektury zaměřené na doménu,
+také nazývané \emph{Domain-centric architectures},
+je produktem vývoje z předchozího typu architektur.
+Střed dění je přesunut na aplikační vrstvu
+--- jak jde vidět z obrázku \ref{fig:architecture_domain} ---,
```

Comment:
`---` je používáno velmi často a ne úplně vhodně, když je to vsuvka a hned za tím zase čárka jako zde. Vhodnější by bylo použít čárky nebo závorky.

(Navíc si nejsem úplně jistý zda má být pro vsuvku `---` nebo `--`)

Reply:
V češtině je pro vsuvku ok --- i --, ale používat konzistentně. (v angličtině --- bez mezer)

Nastuduji si ještě pravidla a rozdíly mezi vsuvkami pomocí pomlček, čárek a závorek. Děkuji. :-)

Reply:
> Vedle krátké pomlčky (–⁠) se v textu někdy používá také pomlčka dlouhá (—). Použití znaků pomlčky a dlouhé pomlčky musí být jednotné, v jednom dokumentu lze používat pomlčku (–⁠) pro označení rozsahu nebo nahrazení výrazů a či versus a dlouhou pomlčku (—) v ostatních případech, např. Internetovou jazykovou příručku — oficiální webovou stránku jazykové poradny Ústavu pro jazyk český AV ČR — navštívilo v letech 2009–⁠2018 více než dva a půl milionu uživatelů.

https://prirucka.ujc.cas.cz/?id=165